titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/sanitizer/JobAssertions.java

```diff
@@ -56,6 +63,20 @@ public boolean isValidIamRole(String iamRole) {
         return !StringExt.safeTrim(iamRole).isEmpty();
     }
 
+    public boolean isEnvironmentNotTooLarge(Map<String, String> environment) {
+        if(CollectionsExt.isNullOrEmpty(environment)) {
+            return true;
+        }
+
+        int totalSize = environment.entrySet().stream().mapToInt(entry -> {
+            int keySize = StringExt.isEmpty(entry.getKey()) ? 0 : entry.getKey().getBytes(UTF_8).length;
+            int valueSize = StringExt.isEmpty(entry.getValue()) ? 0 : entry.getValue().getBytes(UTF_8).length;
+            return keySize + valueSize + 2;
```

Comment:
Might be worth adding a comment saying the 2 additional bytes are for the equal sign and the NUL terminator.

Reply:
Done